dm/syncer/syncer.go

```diff
@@ -2240,7 +2240,7 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 		// check pass SafeModeExitLoc and try disable safe mode, but not in sharding or replacing error
 		safeModeExitLoc := s.checkpoint.SafeModeExitPoint()
-		if safeModeExitLoc != nil && shardingReSync == nil {
+		if s.safeMode.Enable() && safeModeExitLoc != nil && shardingReSync == nil {
```

Comment:
your change will cause the problem stated in line 2248 ~ 2252. Please keep old logic or flush FlushSafeModeExitPoint when it's set to nil in safe_mode.go

Reply:
I think it is the safest that exec `FlushSafeModeExitPoint` in the safe_mode.go.

But, a discussion, L2248-L2252 does also not occur without `FlushSafeModeExitPoint when it's set to nil in safe_mode.go`. Because `enableSafeModeInitializationPhase` is only called in the `Syncer.Run()` beginning. At this time, `SafeModeExitPoint` just Load from Syncer.go:456 and Syncer.go:1660. So, it is same as the disk's `SafemodeExitPoint`.